syft/pkg/cataloger/javascript/parse_package_lock.go

```diff
@@ -24,6 +24,34 @@ type packageLock struct {
 	Packages        map[string]lockPackage
 }
 
+// packageLockLicense
+type packageLockLicense []string
+
+func (licenses *packageLockLicense) UnmarshalJSON(data []byte) (err error) {
+	// The license field could be either a string or an array.
+
+	// 1. An array
+	var arr []string
+	if err := json.Unmarshal(data, &arr); err == nil {
+		*licenses = arr
+		return nil
+	}
+
+	// 2. A string
+	var str string
+	if err = json.Unmarshal(data, &str); err == nil {
+		*licenses = make([]string, 1)
+		(*licenses)[0] = str
+		return nil
+	}
+
+	// 3. Unexpected
```

Comment:
maybe add something like `if len(data) > 2 {  // debug log contents }`?

Reply:
added a debug log, did it for `len(data) > 0`. 

Note the log here would not be so informative as it is just the license value and knowing it happended in a `package-lock.json`. Though better than complete silence, sufficent for a first pass fix imo.